tests/test_subdomains.py

```diff
@@ -162,6 +162,66 @@ def define(self, dimensions):
 
         assert u0.data.all() == u1.data.all() == u2.data.all() == u3.data.all()
 
+    sd_specs = [None, ('middle', 1, 7), ('middle', 2, 3), ('middle', 7, 1),
```

Comment:
Need a `('middle', 5, 5)` case for the case it's fully contained in the center domain and make sure you have 3 domains along that dimension